src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -34,15 +64,54 @@ private bool TryExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionC
                 return false;
             }
 
-            var formattingService = document.GetLanguageService<IEditorFormattingService>();
-            if (formattingService == null || !formattingService.SupportsFormatDocument)
+            return ExecuteCommandAsync(document, args.TextView, context).Result;
+        }
+
+        private async Task<bool> ExecuteCommandAsync(Document document, ITextView textView, CommandExecutionContext context)
+        {
+            var docOptions = await document.GetOptionsAsync(context.WaitContext.UserCancellationToken).ConfigureAwait(false);
+
+            if (!docOptions.GetOption(FeatureOnOffOptions.IsCodeCleanupRulesConfigured))
             {
-                return false;
+                ShowGoldBarForCodeCleanupConfiguration(document.Project.Solution.Workspace);
+                await Format(textView, document, selectionOpt: null, context.WaitContext.UserCancellationToken).ConfigureAwait(false);
             }
-
-            using (context.WaitContext.AddScope(allowCancellation: true, EditorFeaturesResources.Formatting_document))
+            else
             {
-                Format(args.TextView, document, null, context.WaitContext.UserCancellationToken);
+
+                using (context.WaitContext.AddScope(allowCancellation: true, EditorFeaturesResources.Formatting_document))
+                {
+                    var cancellationToken = context.WaitContext.UserCancellationToken;
```

Comment:
pull tihs out above all other code.  Replace all usages of

Reply:
fixed